lib/livebook/application.ex

```diff
@@ -216,13 +217,23 @@ defmodule Livebook.Application do
             "You specified LIVEBOOK_TEAMS_NAME, but LIVEBOOK_TEAMS_KEY is missing."
           )
 
-      Livebook.Hubs.set_offline_hub(%Livebook.Hubs.Team{
+      hub = %Livebook.Hubs.Team{
         id: "team-#{name}",
         hub_name: name,
         hub_emoji: "💡",
         teams_key: teams_key,
         org_public_key: public_key
-      })
+      }
+
+      Livebook.Hubs.set_offline_hub(hub)
+
+      with :error <- Livebook.Hubs.start_offline_hub(hub, encrypted_secrets) do
```

Comment:
So we are running into the same issues from when we tried going into this direction in the past. We are now adding several `get_offline_hub || fetch_hub` clauses.

If the notebook is going to point to the hub, I think we should officially store it as a hub in storage. We can most likely use a fixed ID and then delete it whenever we are in offline mode. The idea is to do something like this:

```elixir
if name && public_key do
  ...
else
  Hub.delete_hub(:offline)
end
```

WDYT?

Reply:
Ah, we cannot change the ID. Perhaps we store a single key in the storage with the `offline_hub_id`, then whenever we boot Livebook, at the top of this function, we see if there is a offline_hub_id, if so, we delete the hub and the `offline_hub_id` key-value.